src/Test/config.jl

```diff
@@ -10,13 +10,14 @@ struct TestConfig
     # MOI.OPTIMAL or MOI.LOCALLY_SOLVED.
     optimal_status::MOI.TerminationStatusCode
     basis::Bool # can get variable and constraint basis status
+    number_type::Type{<:Number}
```

Comment:
Parametrizing TestConfig would get rid of the abstract type here, which is probably desirable